x/eibc/keeper/demand_order.go

```diff
@@ -86,19 +94,22 @@ func (im IBCMiddleware) createDemandOrderFromIBCPacket(ctx sdk.Context, fungible
 		return nil, fmt.Errorf("validate eibc metadata: %w", err)
 	}
 	// Verify the original recipient is not a blocked sender otherwise could potentially use eibc to bypass it
-	if im.BlockedAddr(fungibleTokenPacketData.Receiver) {
+	if k.BlockedAddr(fungibleTokenPacketData.Receiver) {
 		return nil, fmt.Errorf("not allowed to receive funds: receiver: %s", fungibleTokenPacketData.Receiver)
 	}
 	// Calculate the demand order price and validate it,
 	amt, _ := sdk.NewIntFromString(fungibleTokenPacketData.Amount) // guaranteed ok and positive by above validation
 
 	// Get the fee from the memo
 	fee, _ := eibcMetaData.FeeInt() // guaranteed ok by above validation
-
 	if amt.LT(fee) {
 		return nil, fmt.Errorf("fee cannot be larger than amount: fee: %s: amt :%s", fee, fungibleTokenPacketData.Amount)
 	}
 
+	// Get the bridging fee from the amount
+	bridgingFee := k.dack.BridgingFeeFromAmt(ctx, amt)
+	demandOrderPrice := amt.Sub(fee).Sub(bridgingFee)
```

Comment:
isn't it possible that the `demandOrderPrice` can be zero or less? I think the check on line 105: `if amt.LT(fee)` should include the `bridgingFee` as well

Reply:
also, it would be nice to cover this with tests

Reply:
added check.
UT will be handled in https://github.com/dymensionxyz/dymension/issues/916